### DISCUSSION TITLE
error in import

### DISCUSSION THREAD

**Discussion by FendiMohamed** [vercel/next.js#52048] [Show and tell]
Module not found: ESM packages (supports-color) need to be imported. Use 'import' to reference the package instead. https://nextjs.org/docs/messages/import-esm-externals   idk how to fix this error could somoene help me with it

**Comment by kevinstubbs** [comment]
Just in case anybody still runs into this, I recommend checking out this Stackoverflow answer (https://stackoverflow.com/questions/76591038/module-not-found-esm-packages-supports-color-need-to-be-imported-use-import)

**Comment by Mr-Meshky** [comment]
This error occurs because an ESM package (like 'supports-color') is used with 'require', but these libraries should be 'imported'. In newer versions of Next.js and Node.js, ESM is supported by default, but you need to make sure that all packages used are ESM compatible.

I will tell you several ways to solve it, try it

Change to import: If you use require, change it to import.

 For example:

 // Before
`const supportsColor = require('supports-color');`

// After
`import supportsColor from 'supports-color';`

Update packages: If you have an old package that is not compatible with ESM, you can check if there is a newer version of that package that is compatible or not.

Setting next.config.js: If the above changes don't work, you can set next.config.js to load these packages as CommonJS.

 ```
module.exports = {
 Experimental: {
 esmExternals: False
 }
 };
 
```
This setting makes Next.js use require instead of import and temporarily fixes the problem.
